src/ctap/ctap1.rs

```diff
@@ -60,6 +60,24 @@ impl TryFrom<u16> for Ctap1StatusCode {
     }
 }
 
+impl TryFrom<ApduStatusCode> for Ctap1StatusCode {
+    type Error = ();
+
+    fn try_from(apdu_status_code: ApduStatusCode) -> Result<Ctap1StatusCode, ()> {
```

Comment:
Can you say why we convert between those, instead of removing one of the types and only using the other?

Reply:
@kaczmarczyck @jmichelp  not sure if this is the case, but wouldn't APDU be possibly be used by non CTAP-1 scenarios as well?

Reply:
`Ctap1StatusCode` is a practically a subset of `ApduStatusCode` where a few values have a slightly different meaning in the context of Ctap1. E.g. SW_COND_USE_NOT_SATISFIED means in Ctap1 that we're waiting for user presence whereas in generic APDU context it doesn't have a specific meaning.
We can merge them and only keep the error codes in `apdu.rs`. I have a slight preference of having the representation being integers instead of a `ByteArray`.

If that makes sense at a higher level, we can even have `CtapStatusCode` being an enum containing both `ApduStatusCode` and `Ctap2StatusCode`. This way all the processing functions return `CtapStatusCode` errors.

Reply:
I certainly agree to the first paragraph. For the `CtapStatusCode` refactoring, how many functions allow interchanging these error codes? I thought they have pretty separate domains?

Reply:
@jmichelp @kaczmarczyck not sure how much it helped but I moved `ApduStatusCode` to use integers instead of `ByteArray`s and did a best-effort match for places `Ctap1StatusCode` was in use. Not familiar with provenance of the codes mentioned in the enums so maybe the mapping wasn't perfect

(commit 29dbff7a40a1207e19fc6f2e0249eebd6b6d5db8 )

Reply:
Thanks for the changes! I left comments on the best-effort. I think generally, we probably want to return the same code, if we don't have a strong reason to change it. There is one custom error code, that should be non problematic to change. All others are usually implemented as the specification demands.